django/db/migrations/state.py

```diff
@@ -258,7 +298,17 @@ def rename_field(self, app_label, model_name, old_name, new_name):
                         new_name if to_field_name == old_name else to_field_name
                         for to_field_name in to_fields
                     ])
-        self.reload_model(app_label, model_name, delay=delay)
+        if self._relations is not None:
+            old_name_lower = old_name.lower()
+            new_name_lower = new_name.lower()
+            for from_model, to_model in self._relations.items():
+                for field_name, field in to_model[model_key]:
+                    if field_name == old_name_lower:
+                        field.name = new_name_lower
+                        self._relations[from_model][model_key].append((new_name_lower, field))
+                        self._relations[from_model][model_key].remove((old_name_lower, field))
```

Comment:
Isn't `self._relations[from_model]` the same as `to_model`, so you can just use that variable? (I mentioned this in my previous batch of comments.) Also, should you be iterating over e.g. `list(to_model[model_key])` since you're modifying the collection while looping? I think the looping logic is slightly different from before your most recents edits, if I remember right.

Reply:
Good catch :+1: Updated.